erdjs/src/smartcontracts/codec/option.ts

```diff
@@ -1,55 +1,55 @@
 import * as errors from "../../errors";
-import { OptionalValue, TypeDescriptor } from "../typesystem";
+import { Type, OptionValue } from "../typesystem";
 import { BinaryCodec } from "./binary";
 
-export class OptionalValueBinaryCodec {
+export class OptionValueBinaryCodec {
     private readonly parentCodec: BinaryCodec;
 
     constructor(parentCodec: BinaryCodec) {
         this.parentCodec = parentCodec;
     }
 
     /**
-     * Reads and decodes an OptionalValue from a given buffer,
+     * Reads and decodes an OptionValue from a given buffer,
      * with respect to: {@link https://docs.elrond.com/developers/developer-reference/the-elrond-serialization-format | The Elrond Serialization Format}. 
      */
-    decodeNested(buffer: Buffer, typeDescriptor: TypeDescriptor): [OptionalValue, number] {
+    decodeNested(buffer: Buffer, type: Type): [OptionValue, number] {
         if (buffer[0] == 0x00) {
-            return [new OptionalValue(), 1];
+            return [new OptionValue(type), 1];
         }
 
-        let [decoded, decodedLength] = this.parentCodec.decodeNested(buffer.slice(1), typeDescriptor);
-        return [new OptionalValue(decoded), decodedLength + 1];
+        let [decoded, decodedLength] = this.parentCodec.decodeNested(buffer.slice(1), type);
```

Comment:
I think we should also check if buffer[0] == 0x01 and return an error otherwise instead of decoding for any non-zero value for the first byte

Reply:
Fixed.